vault/request_handling.go

```diff
@@ -2307,6 +2304,7 @@ func (c *Core) LocalUpdateUserFailedLoginInfo(ctx context.Context, userKey Faile
 		// delete the entry from the map, if no key exists it is no-op
 		delete(c.userFailedLoginInfo, userKey)
 	}
+	c.userFailedLoginInfoLock.Unlock()
```

Comment:
Deferring this is probably a good idea, though I understand the whole change was reverted because of a race. Feels like adding this back would still make sense.